scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/DockerService.java

```diff
@@ -0,0 +1,43 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Service;
+
+@Service
+@PropertySource("classpath:config.properties")
+public class DockerService {
+
+    @Value("${docker.dockerfile.path}")
+    private String defaultDockerfilePath;
+    
+    @Autowired
+    private DockerRestConnection dockerRestConnection;
+    
+    public String buildImage(String dockerFileLocation) throws FileNotFoundException, InterruptedException{
+        File file = new File(dockerFileLocation + ".tar");
+        String response = dockerRestConnection.dockerClient.buildImageCmd(new FileInputStream(file)).start().awaitImageId();
+        return response;
+    }
+   
+    public DockerService buildContainer(String imageId, String[] userArgs) throws IOException{   
+                             
+        String containerId = dockerRestConnection.dockerClient.createContainerCmd(imageId).withCmd(userArgs).exec().getId();
+
+        dockerRestConnection.dockerClient.startContainerCmd(containerId).exec();
+
+        return this;
```

Comment:
why is this method returning `this`? It's a Service in singleton scope, there is just one object.